### PR TITLE
Add Support for OpenAI Assistants in SAML 

### LINKED ISSUE #751: Add support for OpenAI tools and files usage in YAML schema.
### ⚠️ Please check that this feature request hasn't been suggested before.

- [X] I searched previous [Ideas in Discussions](https://github.com/homanp/superagent/discussions/categories/ideas) didn't find any similar feature requests.
- [X] I searched previous [Issues](https://github.com/homanp/superagent/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

Add support for OpenAI tools and files usage in YAML schema.

- If a user chooses to add files we need to create them with the OpenAI SDK: https://platform.openai.com/docs/api-reference/files. We also need to add the retrieval tool to the OAI assistant.

- If  a user add tools (code interpreter, retrieval or custom functions) these need to be added to the assistant as well. 

### 📝 Additional Context

_No response_

### Acknowledgements

- [X] My issue title is concise and descriptive.
- [X] I have searched the existing issues to make sure this feature has not been requested yet.
- [X] I have provided enough information for the maintainers to understand and evaluate this request.

### PR DESCRIPTION
**Pull request by elisalimli**
Fixes #751

### CODE REVIEW

File: libs/superagent/app/api/agents.py [hunk 1]
```diff
@@ -381,9 +375,30 @@ async def update(
...
-
+        if data.openaiMetadata is None:
+            return {"success": True, "data": data}
+        openaiMetadata = json.loads(data.openaiMetadata)
+        if openaiMetadata:
+            llm = await prisma.llm.find_first_or_raise(
+                where={"provider": "OPENAI", "apiUserId": api_user.id}
+            )
+            oai = AsyncOpenAI(api_key=llm.apiKey)
+            oai_assistant = await oai.beta.assistants.update(
```
Comment (elisalimli) [inline libs/superagent/app/api/agents.py:367]: we are already updating openai assistant above, there's no need for this.

**Review by homanp — CHANGES_REQUESTED** [review]
``` 
    async for chunk in self.body_iterator:
  File "/Users/ismailpelaseyed/Projects/superagent/libs/superagent/app/api/workflows.py", line 279, in send_message
    track_agent_invocation(
  File "/Users/ismailpelaseyed/Projects/superagent/libs/superagent/app/utils/analytics.py", line 43, in track_agent_invocation
    "llm_model": data.get("agent", {}).get("llmModel", None),
                 ^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Agent' object has no attribute 'get'
```
Getting this when a invoke a workflow that is created. Just press New Workflow and you will see it.

File: libs/ui/app/agents/[agentId]/chat.tsx [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+      </div>
+      <ScrollArea className="relative flex grow flex-col px-4">
+        <div className="absolute inset-x-0 top-0 z-20 h-20 bg-gradient-to-b from-background from-0% to-transparent to-50%" />
+        {selectedView === "chat" ? (
+          <div className="mb-20 mt-10 flex flex-col space-y-5 py-5">
+            <div className="container mx-auto flex max-w-4xl flex-col">
+              {messages.map((message, index) => (
+                <Message
+                  key={index}
+                  traceId={session ? `${agent.id}-${session}` : agent.id}
```
Comment (homanp) [inline libs/ui/app/agents/[agentId]/chat.tsx:216]: This throws an error in build error:

```

Type error: Type '{ type: string; message: string; isSuccess?: boolean \| undefined; key: number; traceId: string; profile: Profile; }' is not assignable to type 'IntrinsicAttributes & MessageProps'.
--
15:59:30.580 | Property 'traceId' does not exist on type 'IntrinsicAttributes & MessageProps'.
15:59:30.580 |  
15:59:30.580 | 214 \|                 <Message
15:59:30.580 | 215 \|                   key={index}
15:59:30.580 | > 216 \|                   traceId={session ? `${agent.id}-${session}` : agent.id}
15:59:30.580 | \|                   ^
15:59:30.580 | 217 \|                   profile={profile}
15:59:30.580 | 218 \|                   {...message}
15:59:30.580 | 219 \|                 />
15:59:30.881 | Error: Command "npm run build" exited with 1
```

**Review by homanp — CHANGES_REQUESTED** [review]
@elisalimli two things:

1. Stop streaming OpenAI assistnat
2. Updating workflows breaks the workflow steps as per our call
3. The agent chat comment I just added to the PR. Fails on build:
```
Type error: Type '{ type: string; message: string; isSuccess?: boolean \| undefined; key: number; traceId: string; profile: Profile; }' is not assignable to type 'IntrinsicAttributes & MessageProps'.
--
15:59:30.580 | Property 'traceId' does not exist on type 'IntrinsicAttributes & MessageProps'.
15:59:30.580 |  
15:59:30.580 | 214 \|                 <Message
15:59:30.580 | 215 \|                   key={index}
15:59:30.580 | > 216 \|                   traceId={session ? `${agent.id}-${session}` : agent.id}
15:59:30.580 | \|                   ^
15:59:30.580 | 217 \|                   profile={profile}
15:59:30.580 | 218 \|                   {...message}
15:59:30.580 | 219 \|                 />
15:59:30.881 | Error: Command "npm run build" exited with 1
```

**Pull request merged**